### ISSUE TITLE
Allow manual to be compiled without all dependencies being available?

### ISSUE THREAD

**Issue by wilfwilson** [semigroups/Semigroups#805]
(This is similar to #745 but not the same.)

As things stand, a user is required to have all of the dependencies of Semigroups present on the computer in order to compile the manual of Semigroups (although they don't have to be loadable). For example, if I delete Digraphs, then the following happens:
```
$ ../../bin/gap.sh -A makedoc.g
 ┌───────┐   GAP 4.12dev-1198-g11152e8 built on 2022-02-27 19:06:00+0000
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin21.3.0-default64-kv8
 Configuration:  gmp 6.2.1, GASMAN, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.dev, PrimGrp 3.4.1, SmallGrp 1.4.2, TransGrp 3.5
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
─────────────────────────────────────────────────────────────────────────────
Loading  AutoDoc 2022.02.24 (Generate documentation from GAP source code)
by Sebastian Gutsche (https://algebra.mathematik.uni-siegen.de/gutsche/) and
   Max Horn (https://www.quendi.de/math).
with contributions by:
   Mohamed Barakat,
   Markus Pfeiffer,
   Øystein Skartsæterhagen,
   Chris Wensley,
   Glen Whitney, and
   Fabian Zickgraf.
Homepage: https://gap-packages.github.io/AutoDoc
Report issues at https://github.com/gap-packages/AutoDoc/issues
─────────────────────────────────────────────────────────────────────────────
#I  Semigroups package is not available. To see further details, enter
#I  SetInfoLevel(InfoPackageLoading,4); and try to load the package again.
Error, List Element: <list>[1] must have an assigned value in
  PackageInfo( Pkg[1] ) at makedoc.g:80 called from
<function "unknown">( <arguments> )
 called from read-eval loop at makedoc.g:83
type 'quit;' to quit to outer loop
brk>
```
I think this is not ideal and should be avoided if possible.

The culprit is:
https://github.com/semigroups/Semigroups/blob/8b305924bb0a128bb7e8eeea0c02eeec8acab1b5/makedoc.g#L78-L83
which extracts info from the `PackageInfo.g` file of the dependency via `PackageInfo` and `PackageEntity` (which may not be present).

Line 80 would be easy to work around, just use `UppercaseString(Pkg[1])` directly. But I do not see how to easily work around what you do on line 82. My workaround in Vole is to add each dependency to the bibliography of my manual, including URL, and to manually add `<Cite>` commands at appropriate points when mentioning certain dependencies. But that's less neat, I realise.

If you don't want to fix this @james-d-mitchell, then feel free to close this issue. I realise that it's niche.

**Comment by wilfwilson** [comment]
(#745 caused difficulties when using ReleaseTools, but I think this issue doesn't cause any such trouble).

**Comment by james-d-mitchell** [comment]
Thanks @wilfwilson this is a good suggestion, I'm happy to make the changes, although I might just opt for simply using the package's name and not including the link if it's not available. Also I make the release using `ReleaseTools`, which was pretty much pain free thanks to the `.release` script!

**Comment by wilfwilson** [comment]
Yep, it's very handy!

**Comment by james-d-mitchell** [comment]
Resolved in v4.0.1
